projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/services/cases.service.ts

```diff
@@ -107,8 +107,11 @@ export class CasesService {
 
     const artificialDelay: number = this.appConfig.getTimeoutsCaseRetrievalArtificialDelay();
     const timeoutPeriods = this.appConfig.getTimeoutsForCaseRetrieval();
-    if (timeoutPeriods && timeoutPeriods.length > 0 && timeoutPeriods[1] > 0) {
+    console.log(`Timeout periods: ${timeoutPeriods} seconds.`);
+    if (timeoutPeriods && timeoutPeriods.length > 0 && timeoutPeriods[0] > 0) {
       http$ = this.retryUtil.pipeTimeoutMechanismOn(http$, artificialDelay, timeoutPeriods);
+    } else {
+      console.warn('Skipping to pipe a retry mechanism!');
```

Comment:
Not quite sure what this message means, and I don't think it's really a warning. If we fix the underlying problem we might want to disable the retry. Maybe the message should just be "retry mechanism not enabled" ?

Reply:
Well, we really should keep this retry mechanism for its general purpose. Otherwise, any momentary network glitch will make the application run into the same indefinite spinning wheel issue.